cmd/abapEnvironmentPullGitRepo.go

```diff
@@ -0,0 +1,262 @@
+	bodyText, err := ioutil.ReadAll(resp.Body)
+	json.Unmarshal(bodyText, &body)
+	if body.D == (abapEntity{}) {
+		log.Entry().WithField("StatusCode", resp.Status).WithField("repositoryName", config.RepositoryName).Error("Could not pull the Repository / Software Component")
+		var err = errors.New("Request to ABAP System not successful")
+		return uriConnectionDetails, err
+	}
+	uriConnectionDetails.URL = body.D.Metadata.URI
+	return uriConnectionDetails, nil
+}
+
+func getAbapCommunicationArrangementInfo(config abapEnvironmentPullGitRepoOptions, r runner) (connectionDetailsHTTP, error) {
+
+	var connectionDetails connectionDetailsHTTP
+	var error error
+
+	if config.Host != "" {
+		// Host, User and Password are directly provided
+		connectionDetails.URL = "https://" + config.Host + "/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Pull"
+		connectionDetails.User = config.User
+		connectionDetails.Password = config.Password
+	} else {
+		if config.CfAPIEndpoint == "" || config.CfOrg == "" || config.CfSpace == "" || config.CfServiceInstance == "" || config.CfServiceKey == "" {
+			var err = errors.New("Parameters missing. Please provide EITHER the Host of the ABAP server OR the Cloud Foundry ApiEndpoint, Organization, Space, Service Instance and a corresponding Service Key for the Communication Scenario SAP_COM_0510")
+			return connectionDetails, err
+		}
+		// Url, User and Password should be read from a cf service key
+		var abapServiceKey, error = readCfServiceKey(config, r)
+		if error != nil {
+			log.Entry().Error(error)
+			return connectionDetails, error
+		}
+		connectionDetails.URL = abapServiceKey.URL + "/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Pull"
+		connectionDetails.User = abapServiceKey.Abap.Username
+		connectionDetails.Password = abapServiceKey.Abap.Password
+	}
+	return connectionDetails, error
+}
+
+func readCfServiceKey(config abapEnvironmentPullGitRepoOptions, r runner) (serviceKey, error) {
+
+	var abapServiceKey serviceKey
+
+	// Logging into the Cloud Foundry via CF CLI
+	log.Entry().WithField("cfApiEndpoint", config.CfAPIEndpoint).WithField("cfSpace", config.CfSpace).WithField("cfOrg", config.CfOrg).WithField("User", config.User).Info("Cloud Foundry parameters: ")
+	var cfLoginScript = "cf login -a " + config.CfAPIEndpoint + " -u " + config.User + " -p " + config.Password + " -o " + config.CfOrg + " -s " + config.CfSpace
+	cflogin, error := r.run(cfLoginScript)
+	// cflogin, error := exec.Command("sh", "-c", cfLoginScript).Output()
+	fmt.Printf("%s\n\n", cflogin)
+	if error != nil {
+		log.Entry().Error("Login at cloud foundry failed.")
+		return abapServiceKey, error
+	}
+
+	// Reading the Service Key via CF CLI
+	log.Entry().WithField("cfServiceInstance", config.CfServiceInstance).WithField("cfServiceKey", config.CfServiceKey).Info("Reading service key of service instance...")
+	var cfReadServiceKeyScript = "cf service-key " + config.CfServiceInstance + " " + config.CfServiceKey + " | awk '{if(NR>1)print}'"
```

Comment:
if represented as a command it could also be run on a non-linux developer machine ...

I don't see a reason for creating a dependency to a shell call here ...